crates/sol-macro/src/expand/contract.rs

```diff
@@ -200,6 +204,137 @@ pub(super) fn expand(cx: &ExpCtxt<'_>, contract: &ItemContract) -> Result<TokenS
         }
     });
 
+    let rpc = rpc.then(|| {
+        let contract_name = name;
+        let name = format_ident!("{contract_name}Instance");
+        let name_s = name.to_string();
+        let methods = functions.iter().map(|f| call_builder_method(f, cx));
+        let new_fn_doc = format!(
+            "Creates a new wrapper around an on-chain [`{contract_name}`](self) contract instance.\n\
+             \n\
+             See the [wrapper's documentation](`{name}`) for more details."
+        );
+        let struct_doc = format!(
+            "A [`{contract_name}`](self) instance.\n\
+             \n\
+             Contains type-safe methods for interacting with an on-chain instance of the\n\
+             [`{contract_name}`](self) contract located at a given `address`, using a given\n\
+             provider `P`.\n\
+             \n\
+             See the [module-level documentation](self) for all the available methods."
+        );
+        let (deploy_fn, deploy_method) = bytecode.is_some().then(|| {
+            let doc_str = "todo";
```

Comment:
to be addressed in this PR or a follow up?

Reply:
Here